logs/handler.go

```diff
@@ -0,0 +1,431 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/openfaas/faas-provider/httputils"
+)
+
+var upgrader = websocket.Upgrader{} // use default options
+
+// Requestor submits queries the logging system.
+// This will be passed to the log handler constructor.
+type Requestor interface {
```

Comment:
I hear `requester` is more common in spoken English. https://english.stackexchange.com/questions/29254/whats-the-difference-between-requester-and-requestor

Reply:
I have renamed it